craft/capctl/src/capctl.scala

```diff
@@ -57,43 +58,33 @@ class NcapctlTop(c: NcapctlTopParams)(implicit p: Parameters) extends NcapctlTop
     ioBridgeSink.bundle.reset_n := !(reset.toBool)
 
     // connect ioBridge source and sink
-    ioBridgeSource.bundle.odata   := ioBridgeSink.bundle.odata
+
+    ioNode.bundle.odata := ioBridgeSink.bundle.odata
   }
 
 }
 
 object NcapctlTop {
   def attach(c: NcapctlTopParams)(bap: BlockAttachParams): NcapctlTop = {
     val capctl = NcapctlTopBase.attach(c)(bap)
+    val pwm = NPWMTopBase.attach(NPWMTopParams.defaults(0x70000L, c.blackbox.cacheBlockBytes))(bap)
 
-    // User code here
     implicit val p: Parameters = bap.p
 
-//------------------------------------------------------------------
-    // instantiate and connect the capctl
-    // bap.testHarness {
-    {
-
-      // route capctl signals to the testharness
-      val capctlNode = BundleBridgeSink[NcapctlTopIO]()
-      capctlNode := capctl.ioBridgeSource
-
-      // route PWM signals to the testharness
-      // val PWMNode = BundleBridgeSink[NPWMTopIO]()
-      // PWMNode := NPWMTop.PWM.ioBridgeSource
-
-      // connect the PWM and capctl signals
-      InModuleBody {
-          // PWMNode.bundle.capt0_event     := capctlNode.bundle.odata(0) 
-          // PWMNode.bundle.capt1_event     := capctlNode.bundle.odata(1) 
-          // PWMNode.bundle.capt2_event     := capctlNode.bundle.odata(2) 
-          // PWMNode.bundle.capt3_event     := capctlNode.bundle.odata(3) 
-          // PWMNode.bundle.capt4_event     := capctlNode.bundle.odata(4) 
-          // PWMNode.bundle.capt5_event     := capctlNode.bundle.odata(5) 
-      }
-    }
+    // connect the PWM and capctl signals
+    val capctlNode = BundleBridgeSink[NcapctlTopIO]
+    capctlNode := capctl.ioNode
+
+    val pwmNode = BundleBridgeSink[NPWMTopIO]
+    pwmNode := pwm.ioNode
```

Comment:
you also need to create a `BundleBridgeSink` to route the `PWM` signals to the subsystem.